eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Deleg.hs

```diff
@@ -0,0 +1,214 @@
+          !> To mCoin
+      DRepAlreadyRegisteredForStakeKeyDELEG stakeCred ->
+        Sum (DRepAlreadyRegisteredForStakeKeyDELEG @era) 5
+          !> To stakeCred
+      WrongCertificateTypeDELEG ->
+        Sum (WrongCertificateTypeDELEG @era) 6
+
+instance Era era => DecCBOR (ConwayDelegPredFailure era) where
+  decCBOR = decode $ Summands "ConwayDelegPredFailure" $ \case
+    1 -> SumD IncorrectDepositDELEG <! From
+    2 -> SumD StakeKeyAlreadyRegisteredDELEG <! From
+    3 -> SumD StakeKeyNotRegisteredDELEG <! From
+    4 -> SumD StakeKeyHasNonZeroAccountBalanceDELEG <! From
+    5 -> SumD DRepAlreadyRegisteredForStakeKeyDELEG <! From
+    6 -> SumD WrongCertificateTypeDELEG
+    n -> Invalid n
+
+newtype ConwayDelegEvent era = DelegEvent (Event (EraRule "DELEG" era))
+
+instance
+  ( EraPParams era
+  , State (EraRule "DELEG" era) ~ DState era
+  , Signal (EraRule "DELEG" era) ~ ConwayDelegCert (EraCrypto era)
+  , Environment (EraRule "DELEG" era) ~ DelegEnv era
+  , EraRule "DELEG" era ~ ConwayDELEG era
+  ) =>
+  STS (ConwayDELEG era)
+  where
+  type State (ConwayDELEG era) = DState era
+  type Signal (ConwayDELEG era) = ConwayDelegCert (EraCrypto era)
+  type Environment (ConwayDELEG era) = DelegEnv era
+  type BaseM (ConwayDELEG era) = ShelleyBase
+  type PredicateFailure (ConwayDELEG era) = ConwayDelegPredFailure era
+  type Event (ConwayDELEG era) = ConwayDelegEvent era
+
+  transitionRules = [conwayDelegTransition @era]
+
+conwayDelegTransition :: forall era. EraPParams era => TransitionRule (ConwayDELEG era)
+conwayDelegTransition = do
+  TRC
+    ( DelegEnv _slot _ptr _acnt pp
+      , dState@DState {dsUnified}
+      , c
+      ) <-
+    judgmentContext
+  let pd = pp ^. ppKeyDepositL
+  case c of
+    ConwayRegCert stakeCred sMayDeposit -> do
+      checkDepositAgainstPParams pd sMayDeposit
+      checkStakeKeyNotAlreadyRegistered stakeCred dsUnified
+      pure $ dState {dsUnified = acceptDepositForStakeKey stakeCred dsUnified pd}
+    ConwayUnRegCert stakeCred sMayDeposit -> do
+      checkStakeKeyIsAlreadyRegistered stakeCred dsUnified
+      checkStakeKeyHasZeroBalance stakeCred dsUnified
+      unless (sMayDeposit == SNothing) $ checkDepositAgainstPaidDeposit stakeCred dsUnified sMayDeposit
+      let umRDRemoved = Set.singleton stakeCred UM.⋪ UM.RewDepUView dsUnified
+          umSPoolRemoved = Set.singleton stakeCred UM.⋪ UM.SPoolUView umRDRemoved
+          newUMap = UM.PtrUView umSPoolRemoved UM.⋫ Set.singleton stakeCred -- Although we don't care about Ptrs in Conway, we remove them when we can.
+      pure $ dState {dsUnified = newUMap}
+    ConwayDelegCert stakeCred delegatee -> do
+      checkStakeKeyIsAlreadyRegistered stakeCred dsUnified
+      pure $
+        dState
+          { dsUnified = processDelegation stakeCred delegatee dsUnified
+          }
+    ConwayRegDelegCert stakeCred delegatee coin -> do
+      coin
+        == pd
+        ?! IncorrectDepositDELEG (SJust coin)
```

Comment:
Putting it all on one line is a bit more readable. See my other comment about the SMaybe
```suggestion
      coin == pd ?! IncorrectDepositDELEG coin
```

Reply:
Yes! :100: Thanks!